pkg/controller/servicecatalog/provisionedproduct/lifecycle.go

```diff
@@ -192,3 +155,25 @@ func (c *custom) preDelete(_ context.Context, cr *svcapitypes.ProvisionedProduct
 	obj.ProvisionedProductName = aws.String(meta.GetExternalName(cr))
 	return false, nil
 }
+
+func setConditions(describeRecordOutput *svcsdk.DescribeRecordOutput, resp *svcsdk.DescribeProvisionedProductOutput, cr *svcapitypes.ProvisionedProduct) {
```

Comment:
This func still can't pass the linter, I will fix it tomorrow

Reply:
The function looks ok, it is well defined and, yes, it has many cases, but it is consistent in what it does.
So I suggest to not break it or condensate too much the code just for the sake of cyclomatic complexity (it is 11 ...) because the readability will be worst in the end.